nodeJS/express/views.md

```diff
@@ -0,0 +1,165 @@
+
+In order to work with variables, we use the `<%=` tag. This lets us convert variables into values when the application is loaded. Otherwise, stick with `<%` when using JavaScript logic.
+
+Here's a quick example that makes includes arrays and loop logic.
+
+```js
+<% const animals = ["Cat", "Dog", "Lemur", "Hawk"] %>
+
+<% animals.map(animal => { %>
+- <%= animal%>s are cute <% }) %>
+```
+
+### Using EJS with Express
+
+Let's use `EJS` with `Express`. First, create an `EJS` template file called `index.ejs`, and add the following:
+
+```html
+<html>
+  <body>
+    <%= message %>
+  </body>
+</html>
+```
+
+And in app.js, render this template file in one of your routes:
+
+```js
+app.get("/", (req, res) => {
+  res.render("index", { message: "EJS rocks!" });
+});
+```
+
+When the `"/"` route is matched, the template file matching `"index"` is rendered with the message value passed, such that you should see:
+
+```
+EJS rocks!
+```
+
+### Reusable web components
+
+You may want to include webpage components that are shared across different pages, such as a sidebar or a header. To insert such components into your pages, we make use of the `include` command. This requires the name of the file to be inserted, and optionally an object of data you wish to pass.
+
+Say you have the following navbar component called `"navbar.ejs"`:
+
+```html
+<!-- navbar.ejs -->
+<nav>
+  <ul>
+    <% for (let i = 0; i < links.length; i++) { %>
+    <li>
+      <a href="<%= links[i].href %>">
+        <span> <%= links[i].text %> </span>
+      </a>
+    </li>
+    <% } %>
+  </ul>
+</nav>
+```
+
+You can insert this component into another EJS file like so:
+
+```html
+<!-- index.ejs -->
+<html>
+  <head>
+    <title>Homepage</title>
+    <body>
+      <%- include('navbar', {links: links}) %>
+    </body>
+  </head>
+</html>
+```
+
+This can be used to include headers and footers in all of your pages, for example.
+
+Note that the navbar expects a `links` value. To pass this data into the navbar, you can pass it when rendering `index.ejs` which contains the navbar. Modify `app.js` such that a `links` object is defined and passed into the `render` function in the `"/"` route handler:
+
+```js
+// app.js
+const links = [
+  { href: "/", text: "Home" },
+  { href: "about", text: "About" },
+];
+
+app.get("/", (req, res) => {
+  res.render("index", { links: links });
+});
+```
+
+Here's another example of how to use `includes` to dynamically render a list of variables:
+
+```html
+<ul>
+  <% users.map( user => { %> <%- include('user/show', {user: user}) %> <% }) %>
+</ul>
+```
+
+Note the use of the raw output tag `<%-` with the `include` which is used to avoid double-escaping the HTML output.
+
+### Assignment
+
+<div class="lesson-content__panel" markdown="1">
+
+1. A RESOURCE ITEM
+   - AN INSTRUCTION ITEM
+1. A PRACTICE ITEM
+   - A TASK ITEM
+
+</div>
```

Comment:
We should add relevant documentation from the Express site, and a couple of practice assignment items, like: "Add a view for an about page which should render on the `/about` route"

Reply:
I've just added a few assignment items, including a resource from Express regarding template engines (I've addressed that it uses Pug rather than EJS), and to create and add some components. Sorry for the delay!